tests/basic_cxx_qt/rust/src/locking.rs

```diff
@@ -0,0 +1,46 @@
+// SPDX-FileCopyrightText: 2023 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+/// Two QObject that allow for testing that locking works
+#[cxx_qt::bridge(cxx_file_stem = "locking")]
+pub mod ffi {
+    use std::{thread, time::Duration};
+
+    /// A QObject which has cxx_qt::Locking
+    #[cxx_qt::qobject]
+    #[derive(Default)]
+    pub struct RustLockingEnabled {
+        #[qproperty]
+        counter: u32,
+    }
+
+    impl qobject::RustLockingEnabled {
+        #[qinvokable]
+        pub fn increment(self: Pin<&mut Self>) {
+            let counter = self.counter();
+            thread::sleep(Duration::from_millis(100));
+            self.set_counter(counter + 1);
+        }
+    }
+
+    /// A QObject which has !cxx_qt::Locking
+    #[cxx_qt::qobject]
+    #[derive(Default)]
+    pub struct RustLockingDisabled {
+        #[qproperty]
+        counter: u32,
+    }
+
+    unsafe impl !cxx_qt::Locking for RustLockingDisabled {}
```

Comment:
This fails with

```
[build] error[E0198]: negative impls cannot be unsafe
[build]   --> tests/basic_cxx_qt/rust/src/locking.rs:36:17
[build]    |
[build] 36 |     unsafe impl !cxx_qt::Locking for RustLockingDisabled {}
[build]    |     ------      -^^^^^^^^^^^^^^^
[build]    |     |           |
[build]    |     |           negative because of this
[build]    |     unsafe because of this
[build] 
[build] error[E0658]: negative trait bounds are not yet fully implemented; use marker types for now
[build]   --> tests/basic_cxx_qt/rust/src/locking.rs:36:17
[build]    |
[build] 36 |     unsafe impl !cxx_qt::Locking for RustLockingDisabled {}
[build]    |                 ^^^^^^^^^^^^^^^^
[build]    |
[build]    = note: see issue #68318 <https://github.com/rust-lang/rust/issues/68318> for more information
```

Reply:
Solved this needed to be parsed by our macro, which needed `qobject::RustLockingDisabled` instead.